### PR TITLE
feat: add experimental `db0` driver

### PR DESCRIPTION
**Pull request by zsilbi**
Hello!

Resolves https://github.com/unjs/unstorage/issues/400

I added tests for `libsql` (local file) and `postgresql` (only runs when the connection string in the `.env` is present.
And I also updated the docs.

Notes:
<details>
<summary>Inlining db0 in the vitest config was necessary because of this error from the postgres test</summary>
<pre>
FAIL  test/drivers/db0.test.ts [ test/drivers/db0.test.ts ]
SyntaxError: Named export 'Client' not found. The requested module 'pg' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'pg';
const { Client } = pkg;
</pre>
</details>

Related: https://github.com/unjs/db0/issues/62 https://github.com/unjs/db0/pull/63

### CODE REVIEW

**Comment by pi0** [comment]
Thanks for the PR dear @zsilbi it is a good start.

TBH i have to go over it for next steps (and we need a newer db0 release since it is stall in current experimental version) but some pointers if you can help in the meantime:

- We need to most importantly support `sqlite` dialect for db0 and have it as default
- Table creation should be automated too so that this driver can be used in zero config setups (we might need to have migration too later as a note)
- (chore) i would remove dotenv requirement. pglite and sqlite can be used for testing

File: src/drivers/db0.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+const DRIVER_NAME = "db0";
+
+export default defineDriver((opts: Db0DriverOptions) => {
+  opts.table = opts.table || "storage";
+
+  const dialects: Partial<Record<Dialect, Partial<Driver>>> = {
+    mysql: {
+      setItem: async (key, value) => {
+        await opts.database
```
Comment (pi0) [inline src/drivers/db0.ts:30]: Refactor: To improve maintenance, i would use an if condition inside main `setItem` to override behavior for `mysql` dialect

**Comment by zsilbi** [comment]
>  Table creation should be automated too so that this driver can be used in zero config setups

Should it always try to create the table before the first query, or only after the first failed query and then retry?

To make it more easier to configure, `dialect` entries could also be added to `db0` connectors.
The database instance could return the dialect of the provided connector so this driver could use that.

**Comment by pi0** [comment]
>  or only after the first failed query and then retry?

i wasn't thinking of it sounds a nice idea!

> To make it easier to configure, dialect entries could also be added to db0 connectors.

i was thinking of it actually! feel free to open an issue in db0 to support this.

**Comment by zsilbi** [comment]
> > or only after the first failed query and then retry?
> 
> i wasn't thinking of it sounds a nice idea!

I got this working. It also stores the promise of the `CREATE TABLE` query, so concurrent queries could queue up waiting for the same promise to be fulfilled.

> I would remove dotenv requirement. pglite and sqlite can be used for testing

I'll change it to use pglite when it will be available: https://github.com/unjs/db0/pull/110

**Pull request merged**